shared/substrate/testing/events_test.go

```diff
@@ -0,0 +1,74 @@
+package subtest
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ChainSafe/ChainBridge/e2e/substrate"
+	utils "github.com/ChainSafe/ChainBridge/shared/substrate"
+	gsrpc "github.com/centrifuge/go-substrate-rpc-client"
+	"github.com/centrifuge/go-substrate-rpc-client/types"
+)
+
+func TestChain_Events(t *testing.T) {
```

Comment:
Convenient function to verify event definition + types against any running chain. I was able to verify all events in all blocks since genesis, after fixing some of the types in both GSRPC and this repo.
Turn out very handy function, but not sure if we can extract it to a different place. Will require some manual changes to point to the desired network and from/to desired blocks. Comments added explain it.